core/arch/arm/kernel/boot.c

```diff
@@ -1388,14 +1388,18 @@ void __weak boot_save_args(unsigned long a0, unsigned long a1,
 	 * a4	- Not used
 	 */
 
-	if (IS_ENABLED(CFG_TRANSFER_LIST) &&
-	    a1 == (TRANSFER_LIST_SIGNATURE | REG_CONVENTION_VER_MASK)) {
-		if (IS_ENABLED(CFG_ARM64_core)) {
+	if (IS_ENABLED(CFG_TRANSFER_LIST)) {
```

Comment:
Please update `Scenario B` above to match the code below.

Reply:
Thanks. I'll update.